### ISSUE TITLE
Remove Bindings

### ISSUE THREAD

**Issue by 4uva4ok1905** [jonataslaw/getx#1735]
Hi, update from **4.1.4** -> **4.3.4**, and I have problem with remove Controllers after Get.offNamed

**4.1.4** -  **Working normal**
FIRST VISIT FAVORITE PAGE
`[GETX] NEW ROUTE /favorite`
`[GETX] Instance "FavoriteController" has been created`
`[GETX] Instance "FavoriteController" has been initialized`

GO TO PROFILE PAGE - **Get.offNamed(Routes.PROFILE);**
`[GETX] REPLACE ROUTE /favorite`
`[GETX] NEW ROUTE /profile`
`[GETX] Instance "ProfileController" has been created`
`[GETX] Instance "ProfileController" has been initialized`
`[GETX] "FavoriteController" onDelete() called`

FROM PROFILE TO FAVORITE PAGE - **Get.offNamed(Routes.FAVORITE);**
`[GETX] REPLACE ROUTE /profile`
`[GETX] NEW ROUTE /favorite`
`[GETX] Instance "FavoriteController" has been created`
`[GETX] Instance "FavoriteController" has been initialized`
`[GETX] "ProfileController" deleted from memory`

**4.3.4**
FIRST VISIT FAVORITE PAGE
`[GETX] NEW ROUTE /favorite`
`[GETX] Instance "FavoriteController" has been created`
`[GETX] Instance "FavoriteController" has been initialized`

GO TO PROFILE PAGE - **Get.offNamed(Routes.PROFILE);**
`[GETX] REPLACE ROUTE /favorite`
`[GETX] NEW ROUTE /profile`
`[GETX] Instance "ProfileController" has been created`
`[GETX] Instance "ProfileController" has been initialized`
**Why FavoriteController not delete?**

FROM PROFILE TO FAVORITE PAGE - **Get.offNamed(Routes.FAVORITE);**
`[GETX] REPLACE ROUTE /profile`
`[GETX] NEW ROUTE /favorite`
**Why ProfileController not delete and not Init FavoriteController?**

**Comment by 4uva4ok1905** [comment]
Minimal working version
```
import 'package:flutter/material.dart';
import 'package:get/get.dart';
import 'routes/app_pages.dart';

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();

  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return GetMaterialApp(
      debugShowCheckedModeBanner: false,
      initialRoute: AppPages.INITIAL,
      getPages: AppPages.routes,
    );
  }
}

class AppPages {
  static const INITIAL = Routes.FAVORITE;

  static final routes = [
    GetPage(
      name: Routes.FAVORITE,
      page: () => FavoriteScreen(),
      binding: FavoriteBinding(),
    ),
    GetPage(
      name: Routes.PROFILE,
      page: () => ProfileScreen(),
      binding: ProfileBinding(),
    ),
  ];
}

class FavoriteBinding extends Bindings {
  @override
  void dependencies() {
    Get.put<FavoriteController>(FavoriteController());
  }
}

class FavoriteController extends GetxController {}

class FavoriteScreen extends GetView<FavoriteController> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Text('FAVORITE'),
          GestureDetector(
            onTap: (){
              Get.offNamed(Routes.PROFILE);
            },
            child: Text('Go to PROFILE'),
          )
        ],
      ),
    );
  }
}

class ProfileBinding extends Bindings {
  @override
  void dependencies() {
    Get.put<ProfileController>(ProfileController());
  }
}

class ProfileController extends GetxController {}

class ProfileScreen extends GetView<ProfileController> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Text('PROFILE'),
          GestureDetector(
            onTap: (){
              Get.offNamed(Routes.FAVORITE);
            },
            child: Text('Go to FAVORITE'),
          )
        ],
      ),
    );
  }
}

```

LOGS
```
Performing hot restart...
Syncing files to device iPhone 8...
Restarted application in 1 342ms.
[GETX] Instance "GetMaterialController" has been created
[GETX] Instance "GetMaterialController" has been initialized
[GETX] GOING TO ROUTE /favorite
[GETX] Instance "FavoriteController" has been created
[GETX] Instance "FavoriteController" has been initialized
[GETX] REPLACE ROUTE /favorite
[GETX] NEW ROUTE /profile
[GETX] Instance "FavoriteController" was restarted.
[GETX] Instance "ProfileController" has been created
[GETX] Instance "ProfileController" has been initialized
[GETX] REPLACE ROUTE /profile
[GETX] NEW ROUTE /favorite
[GETX] Instance "FavoriteController" has been created
[GETX] Instance "FavoriteController" has been initialized
[GETX] REPLACE ROUTE /favorite
[GETX] NEW ROUTE /profile
[GETX] REPLACE ROUTE /profile
[GETX] NEW ROUTE /favorite
[GETX] REPLACE ROUTE /favorite
[GETX] NEW ROUTE /profile
[GETX] REPLACE ROUTE /profile
[GETX] NEW ROUTE /favorite
```

**Comment by 4uva4ok1905** [comment]
DOCTOR
```
[✓] Flutter (Channel dev, 2.5.0-1.0.pre, on macOS 11.4 20F71 darwin-x64, locale ru-UA)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.58.2)
[✓] Connected device (2 available)
```

**Comment by jonataslaw** [comment]
It appears that in this specific version, using put (instead of lazyPut) causes problems with Get.off*
I'll fix this, but you should be able to run the code without bugs if you use lazyPut in your Binding

```dart
class ProfileBinding extends Bindings {
  @override
  void dependencies() {
    Get.lazyPut<ProfileController>(()=> ProfileController());
  }
}

```

**Comment by 4uva4ok1905** [comment]
I don’t think so
```
import 'package:flutter/material.dart';
import 'package:get/get.dart';

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();

  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return GetMaterialApp(
      debugShowCheckedModeBanner: false,
      initialRoute: AppPages.INITIAL,
      getPages: AppPages.routes,
    );
  }
}


class Routes {
  static const FAVORITE = '/favorite';
  static const PROFILE = '/profile';
}

class AppPages {
  static const INITIAL = Routes.FAVORITE;

  static final routes = [
    GetPage(
      name: Routes.FAVORITE,
      page: () => FavoriteScreen(),
      binding: FavoriteBinding(),
    ),
    GetPage(
      name: Routes.PROFILE,
      page: () => ProfileScreen(),
      binding: ProfileBinding(),
    ),
  ];
}

class FavoriteBinding extends Bindings {
  @override
  void dependencies() {
    Get.lazyPut<FavoriteController>(()=> FavoriteController());
  }
}

class FavoriteController extends GetxController {
  final title = 'Favorite';
}

class FavoriteScreen extends GetView<FavoriteController> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Text(controller.title),
          GestureDetector(
            onTap: (){
              Get.offNamed(Routes.PROFILE);
            },
            child: Text('Go to PROFILE'),
          )
        ],
      ),
    );
  }
}

class ProfileBinding extends Bindings {
  @override
  void dependencies() {
    Get.lazyPut<ProfileController>(()=> ProfileController());
  }
}

class ProfileController extends GetxController {
  final title = 'Profile';
}

class ProfileScreen extends GetView<ProfileController> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Text(controller.title),
          GestureDetector(
            onTap: (){
              Get.offNamed(Routes.FAVORITE);
            },
            child: Text('Go to FAVORITE'),
          )
        ],
      ),
    );
  }
}

```

```
Performing hot restart...
Syncing files to device iPhone 8...
Restarted application in 1 185ms.
[GETX] Instance "GetMaterialController" has been created
[GETX] Instance "GetMaterialController" has been initialized
[GETX] GOING TO ROUTE /favorite
[GETX] Instance "FavoriteController" has been created
[GETX] Instance "FavoriteController" has been initialized
[GETX] REPLACE ROUTE /favorite
[GETX] NEW ROUTE /profile
[GETX] Instance "FavoriteController" was restarted.
[GETX] Instance "ProfileController" has been created
[GETX] Instance "ProfileController" has been initialized
[GETX] REPLACE ROUTE /profile
[GETX] NEW ROUTE /favorite
[GETX] Instance "FavoriteController" has been created
[GETX] Instance "FavoriteController" has been initialized
[GETX] REPLACE ROUTE /favorite
[GETX] NEW ROUTE /profile
[GETX] REPLACE ROUTE /profile
[GETX] NEW ROUTE /favorite
[GETX] REPLACE ROUTE /favorite
[GETX] NEW ROUTE /profile
[GETX] REPLACE ROUTE /profile
[GETX] NEW ROUTE /favorite
[GETX] REPLACE ROUTE /favorite
[GETX] NEW ROUTE /profile
[GETX] REPLACE ROUTE /profile
[GETX] NEW ROUTE /favorite

```

**Comment by jonataslaw** [comment]
Fixed on get 4.3.5
